FOGL-3554: plugin_reason includes tigger time and asset information

pintomax opened the pull request:
FOGL-3554: plugin_reason includes tigger time and asset information

This change needs branch FOGL-3554 of foglamp-service-notification

Example output with two asset configured in a rule

`{ "reason": "cleared", "asset": ["randomwalk", "sinusoid"], "timestamp": "2019-12-31 15:53:01.263673+00:00" }`

pintomax committed:
FOGL-3554: plugin_reason includes tigger time and asset information

FOGL-3554: plugin_reason includes tigger time and asset information

This change needs branch FOGL-3554 of foglamp-service-notification

MarkRiddoch commented on plugin.cpp:
```diff
@@ -308,10 +308,14 @@ bool plugin_eval(PLUGIN_HANDLE handle,
-	ret += "\" }";
+	ret += info.getState() == BuiltinRule::StateTriggered ? "triggered" : "cleared";
+	ret += "\"";
+	ret += ", \"asset\": " + info.getAssets() + ", \"timestamp\": \"" + info.getUTCDateTime() + "\"";
```
Timestamp should come from the reading that triggered the notification

pintomax committed:
FOGL-3554: updated timestamp property in plugin_reason

FOGL-3554: updated timestamp property in plugin_reason
Fix for arrayType
Fix for all assets evaluatio
